examples/python/model-chat.py

```diff
@@ -36,60 +37,25 @@ def main(args):
-        if args.chat_template.count('{') != 1 or args.chat_template.count('}') != 1:
-            raise ValueError("Chat template must have exactly one pair of curly braces with input word in it, e.g. '<|user|>\n{input} <|end|>\n<|assistant|>'")
-    else:
-        if model_type.startswith("phi4"):
-            args.chat_template = '<|im_start|>user<|im_sep|>\n{input}<|im_end|>\n<|im_start|>assistant<|im_sep|>'
-        elif model_type.startswith("phi"): # For Phi2 and Phi3
-            args.chat_template = '<|user|>\n{input} <|end|>\n<|assistant|>'
-        elif model_type.startswith("llama"):
-            args.chat_template = '<|start_header_id|>user<|end_header_id|>\n{input}<|eot_id|><|start_header_id|>assistant<|end_header_id|>'
-            print("Using Chat Template for LLAMA 3, if you are using LLAMA  2 please pass the argument --chat_template '{input} [/INST]')")
-        elif model_type.startswith("qwen2"):
-            args.chat_template = '<|im_start|>user\n{input}<|im_end|>\n<|im_start|>assistant\n'
-        elif model_type == "gemma3_text":
-            args.chat_template = '<start_of_turn>user\n{system_prompt}{input}<end_of_turn>\n<start_of_turn>model\n'
-        else:
-            raise ValueError(f"Chat Template for model type {model_type} is not known. Please provide chat template using --chat_template")
 
     if args.verbose:
         print("Model type is:", model_type)
-        print("Chat Template is:", args.chat_template)
 
     params = og.GeneratorParams(model)
     params.set_search_options(**search_options)
     generator = og.Generator(model, params)
     if args.verbose: print("Generator created")
 
     # Set system prompt
-    if "<|" in args.system_prompt and "|>" in args.system_prompt:
-        # User-provided system template already has tags
-        system_prompt = args.system_prompt
-    else:
-        if model_type.startswith('phi4'):
-            system_prompt = f"<|im_start|>system<|im_sep|>\n{args.system_prompt}<|im_end|>"
-        elif model_type.startswith('phi'): # For Phi2 and Phi3
-            system_prompt = f"<|system|>\n{args.system_prompt}<|end|>"
-        elif model_type.startswith("llama"):
-            system_prompt = f"<|begin_of_text|><|start_header_id|>system<|end_header_id|>\n{args.system_prompt}<|eot_id|>"
-            print("Using System Prompt for LLAMA 3, if you are using LLAMA  2 please pass the argument --system_prompt '<s>[INST] <<SYS>>\\n{args.system_prompt}\\n<</SYS>>')")
-        elif model_type.startswith("qwen2"):
-            system_prompt = f"<|im_start|>system\n{args.system_prompt}<|im_end|>\n"
-        elif model_type == "gemma3_text":
-            system_prompt = f"{args.system_prompt}"
-        else:
-            system_prompt = args.system_prompt
-
-    system_tokens = tokenizer.encode(system_prompt)
+    messages = []
+    messages.append({"role": "system", "content": args.system_prompt})
+    sys_prompt = tokenizer.apply_chat_template(json.dumps(messages))
```

Comment:
Do you recommend to always add `add_generation_prompt`?

Reply:
no not for the `sys_prompt`, but when you want the `<|assistant|>`, so after for the input yes.